### PR TITLE
docs: Add explanation for use of secrets in Google provider

### PR DESCRIPTION
**Pull request by christophersjchow**
Add docs explaining secrets support.

This should be merged only after  https://github.com/serverless/serverless-google-cloudfunctions/pull/287 is merged.

### CODE REVIEW

**Comment by christophersjchow** [comment]
Upstream code has been merged so this is now ready for review @pgrzesik

File: docs/providers/google/guide/functions.md
Comment (pgrzesik) [inline docs/providers/google/guide/functions.md:128]: There's an extra space between `secret` and `environment` words

**Review by pgrzesik — COMMENTED** [review]
I only have one minor remark, otherwise, it looks good 👍

**Review by pgrzesik — APPROVED** [review]
Thanks @christophersjchow 🙇 Let's also wait for a review from @mnapoli

**Review by mnapoli — APPROVED** [review]
Thanks for the contribution!

**Comment by mnapoli** [comment]
Oh I cannot merge as there is a `npm run prettier-check:updated` error.

**Pull request closed without merging**
